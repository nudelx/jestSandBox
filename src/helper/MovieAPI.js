// import axios from 'axios'

// const MovieAPI = {
//   fetch: url => {
//     if (window.fetch) {
//       return axios.get(url)
//         .then( response =>  response)
//         .catch(function (error) {
//           console.log(error);
//         });
//     }
//     return Promise.reject(new Error('oops'))
//   }
// }

const MovieAPI = {
  fetch: url => {
    if (window.fetch) {
      return window.fetch(url)
        .then( resp =>  resp.json())
        .then(resp => resp)
    }
    return Promise.reject(new Error('oops'))
  }
}


export default MovieAPI
