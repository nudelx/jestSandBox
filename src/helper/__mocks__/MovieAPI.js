var movie = {
  title: 'Hope it Will work',
  episode_id: '1_t',
  opening_crawl: 'this is a test',
  director: 'BB-8',
  producer: 'Kylo Ren',
  release_date: '1977-05-25',
  created: '2014-12-10 14:23:31',
  edited: '2015-04-11 09:46:52',
  url: '/this/is/url/to/some/movie/'
}

const movies = page => (movie.episode_id = `test_${page}` && movie)

const MovieAPI = {
  fetch: url => {
    const page = url.split('/').pop()
    return new Promise((resolve, reject) => {
      process.nextTick(
        () =>
          page >= 1 && page <= 3
            ? resolve(movies(page))
            : reject({
                error: 'movie with ' + page + ' not found.'
              })
      )
    })
  }
}

export default MovieAPI
