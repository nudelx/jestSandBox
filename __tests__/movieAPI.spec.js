jest.mock('../src/helper/MovieAPI')
import MovieAPI from '../src/helper/MovieAPI'

describe('testing movie API ', () => {
  it('should return data object with title', () => {
    expect.assertions(1);
    return MovieAPI.fetch('bla/bla/bla/1')
    .then(data => expect(data.title).toEqual('Hope it Will work'));
  })

  it('should return data object with title', () => {
    expect.assertions(1);
    return MovieAPI.fetch('bla/bla/bla/1')
    .then(data => expect(data.director).toEqual('BB-8'));
  })

  it('should match the snapshot', () => {
    expect.assertions(1);
    return MovieAPI.fetch('bla/bla/bla/1')
    .then(data => expect(data).toMatchSnapshot());
  })

  it('should match the error', () => {
    expect.assertions(1);
    return MovieAPI.fetch('bla/bla/bla/5')
    .then(data => expect(data).toMatchSnapshot())
    .catch(error => expect(error).toMatchSnapshot());
  })

})
