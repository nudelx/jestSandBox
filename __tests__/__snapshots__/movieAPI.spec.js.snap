// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing movie API  should match the error 1`] = `
Object {
  "error": "movie with 5 not found.",
}
`;

exports[`testing movie API  should match the snapshot  1`] = `
Object {
  "created": "2014-12-10 14:23:31",
  "director": "BB-8",
  "edited": "2015-04-11 09:46:52",
  "episode_id": [Circular],
  "opening_crawl": "this is a test",
  "producer": "Kylo Ren",
  "release_date": "1977-05-25",
  "title": "Hope it Will work",
  "url": "/this/is/url/to/some/movie/",
}
`;
